cryptoconditions/types/preimage.py

```diff
@@ -0,0 +1,173 @@
+
+    #    Read a fulfillment payload from a Reader and populate this object with that fulfillment.
+
+    #    FULFILLMENT_PAYLOAD =
+    #        VARBYTES PREIMAGE
+
+    #    Args:
+    #        reader (Reader): Source to read the fulfillment payload from.
+    #        payload_size (int): Total size of the fulfillment payload.
+    #    """
+    #    if not isinstance(reader, Reader):
+    #        raise TypeError('reader must be a Reader instance')
+    #    self.preimage = reader.read(payload_size)
+
+    #def write_payload(self, writer):
+    #    """
+    #    Generate the fulfillment payload.
+
+    #    This writes the fulfillment payload to a Writer.
+
+    #    Args:
+    #        writer (Writer): Subject for writing the fulfillment payload.
+    #    """
+    #    if not isinstance(writer, (Writer, Predictor)):
+    #        raise TypeError('writer must be a Writer instance')
+    #    if self.preimage is None:
+    #        raise ValueError('Preimage must be specified')
+
+    #    writer.write(self.preimage)
+    #    return writer
+
+    def to_dict(self):
+        """
+        Generate a dict of the fulfillment
+
+        Returns:
+            dict: representing the fulfillment
+        """
+        return {
+            'type': 'fulfillment',
+            'type_id': self.TYPE_ID,
+            'bitmask': self.bitmask,
+            'preimage': self.preimage.decode()
+        }
+
+    @property
+    def asn1_dict_payload(self):
+        return {'preimage': self.preimage}
+
+    def to_asn1_dict(self):
+        return {PreimageSha256.TYPE_ASN1: self.asn1_dict_payload}
+
+    @property
+    def fingerprint_contents(self):
+        if self.preimage is None:
+            # TODO MissingDataError('Could not calculate hash, no preimage provided')
+            raise Exception('Could not calculate hash, no preimage provided')
+        return self.preimage
+
+    def parse_json(self, data):
+        """
+        Generate fulfillment payload from a dict
+
+        Args:
+            data (dict): description of the fulfillmjson
+        Returns:
+            Fulfillment
+        """
+        self.preimage = urlsafe_b64decode(base64_add_padding(data['preimage']))
+
+    # TODO remove
+    def parse_dict(self, data):
+        """
+        Generate fulfillment payload from a dict
+
+        Args:
+            data (dict): description of the fulfillment
+
+        Returns:
+            Fulfillment
+        """
+        self.preimage = b64decode(data['preimage'])
+
+    def parse_asn1_dict_payload(self, data):
+        self.preimage = data['preimage']
+
+    def calculate_cost(self):
+        if self.preimage is None:
+            raise MissingDataError('Preimage must be specified')
+        return len(self.preimage)
+
+    def validate(self, *args, **kwargs):
+        """
+        Validate this fulfillment.
+
+        For a SHA256 hashlock fulfillment, successful parsing implies that the
```

Comment:
I don't understand this. Don't we need to hash the `preimage` and check that its equal to the condition fingerprint?
From the spec:
> A PREIMAGE-SHA-256 fulfillment is valid iff C.fingerprint is equal to
   the SHA-256 hash digest of F.

Right now I can do:
```python
>> PreimageSha256().validate()
True
```
I don't think that this is correct.